### ISSUE TITLE
Maximum call stack Exceeded

### ISSUE THREAD

**Issue by jpf200124** [ludohenin/gulp-inline-ng2-template#45]
Hey guys,

I have a very large project that I am trying to use with this plugin and I am constantly getting the following error whenever I try to use it.

RangeError: Maximum call stack size exceeded
    at D:\dev\editor\node_modules\gulp-inline-ng2-template\node_modules\async\dist\async.js:842:20
    at D:\dev\editor\node_modules\gulp-inline-ng2-template\node_modules\async\dist\async.js:5218:17
    at D:\dev\editor\node_modules\gulp-inline-ng2-template\node_modules\async\dist\async.js:339:31
    at D:\dev\editor\node_modules\gulp-inline-ng2-template\parser.js:114:13

Based on the discussion [here](http://stackoverflow.com/questions/27327042/rangeerror-call-stack-exceed-on-async-eachseries) I have changed your code locally to use:

 `process.nextTick(cb);`

and everything seems to be working as expected. I can not seem to reproduce the problem on a smaller project. Just wanted to make you aware of the issue and maybe you can investigate further.

Thanks,

**Comment by ludohenin** [comment]
Oh yes, I forgot about that one, `setTimeout` helps clearing the stack. I'll fix that asap.
Thanks for reporting the issue.

**Comment by ps2goat** [comment]
+1

**Comment by ludohenin** [comment]
published as `2.0.2`

**Comment by fcastill** [comment]
I have the same issue, even running `2.0.2`. It happens when using custom style processor.

```
events.js:141
      throw er; // Unhandled 'error' event
      ^
RangeError: Maximum call stack size exceeded
    at next (.../node_modules/async/dist/async.js:812:33)
    at replenish (.../node_modules/async/dist/async.js:864:32)
    at .../node_modules/async/dist/async.js:879:29
    at .../node_modules/async/dist/async.js:842:20
    at .../node_modules/async/dist/async.js:5218:17
    at .../node_modules/async/dist/async.js:339:31
    at .../node_modules/gulp-inline-ng2-template/parser.js:114:13
    at .../node_modules/async/dist/async.js:5213:13
    at .../node_modules/async/dist/async.js:873:21)
    at .../node_modules/async/dist/async.js:879:29
```

If I change the line where it fails (`parser.js:114`) from `cb(null)` to ` process.nextTick(cb)` it works.

**Comment by ludohenin** [comment]
OK, I'll update it. Thanks for the feedback.
I suppose your styleProcessor function isn't async right?

**Comment by fcastill** [comment]
It is async. I actually tried with both async and sync code and both gave me the same error.

**Comment by ludohenin** [comment]
Can you let me know if this is OK now (`v2.0.4`)

**Comment by fcastill** [comment]
Unfortunately I'm still getting the error at `parser.js:114:13`

**Comment by ludohenin** [comment]
arrgghhh, I can't believe it ... I'll dig that down later this week.

**Comment by fcastill** [comment]
Turns out I was processing a 1526LOC js file with no component declaration, that's why it is filling the stack as the `cb(null)` in `parser:114` is called synchronously 1526 times.

My current workaround is to filter only my *.component files to be processed by the inline plugin.

**Comment by jpf200124** [comment]
My problem is identical to what @fcastill describes. Can either use the PR that I submitted or filter out any files that do not have components. Maybe a better solution would be to have some code that ignores any files that do not have components from the start.

**Comment by supamax** [comment]
How it is going? Have the same issue with big project.

**Comment by thegitty** [comment]
Same issue. 

My package.json: "gulp-inline-ng2-template": "^2.0.4",

parser.js, Line 114: cb(null);
After replacing line 114 with 'process.nextTick(cb);' it does work. 

'npm install' seems not to request the newest version. Also my README.md does not list the version 2.0.4.

**Comment by ludohenin** [comment]
published as `2.0.5`
